### PR TITLE
Format all percentages in stacked charts with the same number of decimals

### LINKED ISSUE #9914: Stack 100% bar charts round values inconsistently
@tuaris flagged this problem in another issue: https://github.com/metabase/metabase/issues/6806#issuecomment-485501024

Here's the image they posted.

image

Each value is rounded independently. It makes sense to round 99.52% to 100% on it's own, but not when it's displayed next to 0.48%.

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #9914 

# Current Behavior
stack-percents-current

# New Behavior
stacked-percents-new

# What's the new logic?

Previously all percents were formatted independently. The new code does a first pass to look at all values and determines a fixed number of decimals used to display any value in the chart.

To compute that number of decimals, we format all the the values as percents with a maximum of two significant digits. We use those formatted values to find the maximum number of decimals used in formatted values.

```
0.123456   -> "12%"   (no decimals)
0.00123456 -> "0.12%" (two decimals)
```

If those were the only two values in the chart, the maximum number of decimals would be two and the values would appear like this:

12.00%
0.12%

# Should we do this at all?

I do have a slight preference for the new behavior, but I'm not convinced it adds enough value to justify further complicating this part of the visualization code. Anyone have thoughts?

If we do want to merge this behavior, I'll add tests before marking it ready for review.

### CODE REVIEW

File: frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js [hunk 1]
```diff
@@ -172,6 +172,37 @@ function addPercentSignsToDisplayNames(series) {
   );
 }
 
+// Store a "decimals" property on the column that is normalized
+function addDecimalsToPercentColumn(series, decimals) {
+  return series.map(s =>
+    updateIn(s, ["data", "cols", 1], col => ({ ...col, decimals })),
```
Comment (tlrobinson) [inline frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js]: btw I think `assicIn(s, ["data", "cols", 1, "decimals"], decimals)` is equivalent.

**Review by kdoh — COMMENTED** [review]
I'm a 👍 on this behavior wise. Definitely makes sense to consider the whole context.

**Review by tlrobinson — COMMENTED** [review]
My only concern if there's any performance issues with formatting every number to determine the number of decimals. I can pull it down an benchmark it.

**Comment by tlrobinson** [comment]
Actually I think the performance impact is minimal for any reasonable number of bars.

**Pull request merged**